hack/release.sh

```diff
@@ -32,7 +32,9 @@ function build_release() {
   GOOS=darwin GOARCH=amd64 go build -mod=vendor -ldflags "${ld_flags}" -o ./kn-darwin-amd64 ./cmd/...
   echo "🚧 🎠 Building for Windows"
   GOOS=windows GOARCH=amd64 go build -mod=vendor -ldflags "${ld_flags}" -o ./kn-windows-amd64.exe ./cmd/...
-  ARTIFACTS_TO_PUBLISH="kn-darwin-amd64 kn-linux-amd64 kn-windows-amd64.exe"
+  echo "🚧 🐳 Building the container image"
+  ko publish ${KO_FLAGS} github.com/knative/client/cmd/kn 2>&1 | grep -E '(Loaded|Published)' | rev | cut -d ' ' -f 1 | rev > kn-image-location
+  ARTIFACTS_TO_PUBLISH="kn-darwin-amd64 kn-linux-amd64 kn-windows-amd64.exe kn-image-location"
```

Comment:
This seems weird to me. You really want to publish a text file with a GCR URL in it? Wouldn't it be better to add that to the release notes instead?

Reply:
Goal: Tie the image by hash to the release. 

The published yamls from the other repos, the ones you can `kubectl apply`, accomplish this goal, but that doesn't make sense for `kn`. Once Tekton is further along we could publish a Tekton Task as a yaml file, but I don't think we're ready yet. So just a text file containing the image by hash seems simplest — is there a better way?

Reply:
Initially I used ko resolve command. That generates a yaml file, so that github.com/knative/test-infra/scripts/release.sh can process later in the function tag_images_in_yamls ${ARTIFACTS_TO_PUBLISH} to change the tag.

But later I quit, since tag_images_in_yamls has bug, and it does not parse any image name ko resolve generates.

Reply:
If we can specify the tag here, then we do not need to bother github.com/knative/test-infra/scripts/release.sh. But I do not know how to specify the tag for the image in this release.sh.

Reply:
@hohaichi  tag_images_in_yamls() works fine for all other repos (parsing yamls that `ko resolve` generates), can you please clarify what bug you're experiencing?

@sixolet I understanding that you're interested in the image hash, and that's unrelated to tagging it on GCR. If so, would adding it to the release notes be enough?

Reply:
I'm getting a 404 for the URL you mentioned. Anyway, the regex is not meant to catch image names with hashes. I think you're running an old version of `ko` because newer versions won't append the hash by default, and you don't need to specify -B:

```
$ export KO_DOCKER_REPO=gcr.io/XXXXXXXXX
$ ko publish -P github.com/knative/client/cmd/kn
2019/06/03 10:30:25 Building github.com/knative/client/cmd/kn
2019/06/03 10:30:29 Using base gcr.io/distroless/base:latest for github.com/knative/client/cmd/kn
2019/06/03 10:30:30 Publishing gcr.io/XXXXXXXXX/github.com/knative/client/cmd/kn:latest
2019/06/03 10:30:31 existing blob: sha256:1558143043601a425aa864511da238799b57fcf7d062d47044f6ddd0e04fe99a
2019/06/03 10:30:31 existing blob: sha256:4003b5b92ca98a8926d9112839f3f17e69f4ec4f995abb188a3ce3ccf93cd6d9
2019/06/03 10:30:31 existing blob: sha256:5f5edd681dcbc3a4a9df93e200e59e1708031e65b2299970eabdc91a78cc8234
2019/06/03 10:30:33 pushed blob: sha256:9e0fd63e6f10b3e99b134f68e8f68ba5de196080141d6254d875994c4868030b
2019/06/03 10:30:35 pushed blob: sha256:3ace94f228dcde95346dabc5a0e17c0cff33442a0fd49691d0233d1cfd25e8ed
2019/06/03 10:30:36 gcr.io/XXXXXXXXX/github.com/knative/client/cmd/kn:latest: digest: sha256:4d5a06964d68d645c743d5e02588a8340f749548db47024470ed917d84fc52ce size: 915
2019/06/03 10:30:36 Published gcr.io/XXXXXXXXX/github.com/knative/client/cmd/kn@sha256:4d5a06964d68d645c743d5e02588a8340f749548db47024470ed917d84fc52ce
```

Reply:
OK , as long as it is able to parse the image name, that's fine.

Reply:
will do ko publish ==> ko resolve

Reply:
> @hohaichi tag_images_in_yamls() works fine for all other repos (parsing yamls that `ko resolve` generates), can you please clarify what bug you're experiencing?
> 

@adrcunha I don't have any context.  Did you mean houshengbo? :)

Reply:
Sorry, wrong username from Auto complete. :/